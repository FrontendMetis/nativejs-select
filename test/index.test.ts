import NativejsSelect from '../src/';
import checkCustomSelect from './helpers/checkCustomSelect';
import checkCustomPlaceholder from './helpers/checkCustomPlaceholder';
import checkCustomOptions from './helpers/checkCustomOptions';


describe('Custom select', () => {
  beforeEach(() => document.body.insertAdjacentHTML('beforeend', `
    <select class="customSelect">
      <option value="react" data-icon="img/react.png">React</option>
      <option value="vue" data-icon="img/vue.png">Vue</option>
      <option value="svelte" data-icon="img/svelte.png">Svelte</option>
    </select>
  `));
  
  afterEach(() => document.body.innerHTML = '');

  test('Default mount', () => {
    new NativejsSelect({ selector: '.customSelect' });

    checkCustomSelect();
    checkCustomPlaceholder();
    checkCustomOptions();
  });

  test('Mount with attribute placeholder', () => {
    document
      .querySelector('.customSelect')
      .setAttribute('data-placeholder', 'Placeholder');
    new NativejsSelect({ selector: '.customSelect' });

    checkCustomSelect();
    checkCustomPlaceholder('Placeholder');
    checkCustomOptions();
  });

  test('Mount with option placeholder', () => {
    new NativejsSelect({ 
      selector: '.customSelect', 
      placeholder: 'Placeholder' 
    });

    checkCustomSelect();
    checkCustomPlaceholder('Placeholder');
    checkCustomOptions();
  });
});