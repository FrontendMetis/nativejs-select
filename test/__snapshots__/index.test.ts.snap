// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom select Render default 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    React
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Svelte
      </button>
    </li>
  </ul>
</div>
`;

exports[`Custom select Render with all ui options 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    <span class="nativejs-select__placeholder_fixed">
      fixed: placeholder
    </span>
    Placholder
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        <img src="img/react.png"
             alt="icon"
             style="width: 20px;"
        >
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        <img src="img/vue.png"
             alt="icon"
             style="width: 20px;"
        >
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        <img src="img/svelte.png"
             alt="icon"
             style="width: 20px;"
        >
        Svelte
      </button>
    </li>
  </ul>
</div>
`;

exports[`Custom select Render with custom options 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    <img src="img/react.png"
         alt="icon"
         style="width: 20px;"
    >
    React
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        <img src="img/react.png"
             alt="icon"
             style="width: 20px;"
        >
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        <img src="img/vue.png"
             alt="icon"
             style="width: 20px;"
        >
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        <img src="img/svelte.png"
             alt="icon"
             style="width: 20px;"
        >
        Svelte
      </button>
    </li>
  </ul>
</div>
`;

exports[`Custom select Render with fixed placeholder 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    <span class="nativejs-select__placeholder_fixed">
      fixed: placeholder
    </span>
    React
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Svelte
      </button>
    </li>
  </ul>
</div>
`;

exports[`Custom select Render with fixedPlaceholder and placeholder 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    <span class="nativejs-select__placeholder_fixed">
      fixed: placeholder
    </span>
    Placholder
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Svelte
      </button>
    </li>
  </ul>
</div>
`;

exports[`Custom select Render with placeholder 1`] = `
<div class="nativejs-select">
  <button class="nativejs-select__placeholder">
    Placholder
  </button>
  <ul class="nativejs-select__options">
    <li class="nativejs-select__option"
        data-selected="true"
    >
      <button class="nativejs-select__option_btn">
        React
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Vue
      </button>
    </li>
    <li class="nativejs-select__option">
      <button class="nativejs-select__option_btn">
        Svelte
      </button>
    </li>
  </ul>
</div>
`;
